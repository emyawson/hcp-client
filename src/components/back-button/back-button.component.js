import React from 'react';

import { colors } from 'src/core/styles/colors';
import { Link } from 'src/components';
import { ArrowIcon } from 'src/assets/icons';

import { ArrowLeftSpan, BackButtonContainerDiv } from './back-button.style';

export const BackButton = ({ to }) => (
  <BackButtonContainerDiv>
    <Link to={to}>
      <ArrowLeftSpan>
        <ArrowIcon height={10} fillColor={colors.charcoal} />
      </ArrowLeftSpan>
    </Link>
  </BackButtonContainerDiv>
);
