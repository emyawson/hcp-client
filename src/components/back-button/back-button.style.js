import styled from 'styled-components';

export const ArrowLeftSpan = styled.span`
  transform: rotate(90deg);
  display: inline-block;
`;

export const BackButtonContainerDiv = styled.div`
  padding-right: ${props => props.theme.spacing.two};
  display: inline-block;
`;

ArrowLeftSpan.displayName = 'ArrowLeftSpan';
