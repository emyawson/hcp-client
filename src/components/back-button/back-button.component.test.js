import React from 'react';
import { shallow } from 'enzyme';

import { BackButton } from './back-button.component';

describe('Back button component', () => {
  it('renders properly', () => {
    const props = {
      to: '/prescription',
    };
    const wrapper = shallow(<BackButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
