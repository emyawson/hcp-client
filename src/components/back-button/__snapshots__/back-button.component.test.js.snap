// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back button component renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    to="/prescription"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Link)
        to="/prescription"
      >
        <ArrowLeftSpan>
          <Unknown
            fillColor="#4A4A4A"
            height={10}
          />
        </ArrowLeftSpan>
      </Styled(Link)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <ArrowLeftSpan>
          <Unknown
            fillColor="#4A4A4A"
            height={10}
          />
        </ArrowLeftSpan>,
        "to": "/prescription",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            fillColor="#4A4A4A"
            height={10}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "fillColor": "#4A4A4A",
            "height": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Link)
          to="/prescription"
        >
          <ArrowLeftSpan>
            <Unknown
              fillColor="#4A4A4A"
              height={10}
            />
          </ArrowLeftSpan>
        </Styled(Link)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <ArrowLeftSpan>
            <Unknown
              fillColor="#4A4A4A"
              height={10}
            />
          </ArrowLeftSpan>,
          "to": "/prescription",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              fillColor="#4A4A4A"
              height={10}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "fillColor": "#4A4A4A",
              "height": 10,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
